Sources/WebService/NetworkResourceWithBody.swift

```diff
@@ -157,6 +157,28 @@ public extension NetworkResourceWithBody {
         }
         return self
     }
+
+    /// Sets the HTTP Body with NSData
+    ///
+    /// Internally sets the Content-Type header of the resource to "application/json"
+    ///
+    /// - Parameters:
+    ///   - body: JSON serialised request body
+    /// - Returns: NetworkResourceWithBody
+    @objc @discardableResult func json(body: Data) -> NetworkResourceWithBody {
+
+        ///Checking for creation error
+        guard self.creationError == nil else {
+            return self
+        }
+        ///Sets the content type
+        self.contentType("application/json")
+
+        ///Sets the HTTP Body
+        self.request.httpBody = body
+
+        return self
+    }
```

Comment:
For raw data there is already one function @objc @discardableResult func data(_ data: Data, mimeType: String?)
Do we need this?

Reply:
@abhishekfk Exactly! There's already a function `.data()` to set raw data into the request: https://flipkart.github.io/Swifty/Classes/NetworkResourceWithBody.html#/c:@CM@Swifty@objc(cs)NetworkResourceWithBody(im)data:mimeType: